src/main/java/org/example/controllers/JobRoleController.java

```diff
@@ -65,4 +67,34 @@ public Response getAllJobRoles() {
             return Response.serverError().build();
         }
     }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Returns a list of Job Roles",
+            authorizations = @Authorization(value = HttpHeaders.AUTHORIZATION),
+            response = JobRoleResponse.class,
+            responseContainer = "Filtered list",
+            produces = "application/json")
+    @ApiResponses({
+            @ApiResponse(code = OK, message = "Job roles listed successfully", response = JobRole.class),
+            @ApiResponse(code = INTERNAL_SERVER_ERROR, message = "getAllJobRoles failed, SQL Exception"),
+            @ApiResponse(code = NOT_FOUND, message = "getAllJobRoles failed, DoesNotExistException")
+    })
+    @Path("/filter")
+    public Response getFilteredJobRoles(final @BeanParam JobRoleFilteredRequest jobRoleFilteredRequest) {
+        LOGGER.info("Get filtered job roles request received");
+        try {
+            return Response.ok().entity(jobRoleService.getFilteredJobRoles(jobRoleFilteredRequest)).build();
+        } catch (SQLException e) {
+            LOGGER.error("getFilteredJobRoles failed, SQL Exception\n" + e.getMessage());
```

Comment:
maybe to add some more verbose error? Like "getting of filtered job roles failed due to SQL exception" ? also applies to any other error message in this file